sphinx_gallery/gen_gallery.py

```diff
@@ -365,8 +365,15 @@ def _fill_gallery_conf_defaults(sphinx_gallery_conf, app=None, check_keys=True):
     # XXX anything that can only be a bool (rather than str) should probably be
     # evaluated this way as it allows setting via -D on the command line
     for key in (
+        "download_all_examples",
```

Comment:
Can we get rid of the `XXX` comment above now that you've made these changes?

Reply:
Good idea, I've amended the comment just so we know why we are doing the `_bool_eval`